### DISCUSSION TITLE
GPS telemetry problems ELRS/CRSF related ?

### DISCUSSION THREAD

**Discussion by breadoven** [iNavFlight/inav#7906] [Q&A] [answered]
Can't get GPS telemetry to work correctly using ELRS with EdgeTX (Radiomaster Zorro). Same problem happens on 2 different models with different FCs and receivers so it's not clear what's causing the problem.

HGLRCF722 and OmnibusF4 boards.
Happymodel EP1 and Matek receivers.
Using INAV version 5.0 current master and current ELRS and EdgeTX versions.
Receiver half duplex OFF.
Serial port inversion OFF.

The GPS sensors are initially detected but seemingly immediately fail with the sensor lost error popping up in the radio some time later. The GPS sensor fails because it only ever shows 3 - 4 satellites when the OSD shows many more are available and the `*`update indication on the radio telemetry page disappears for the GPS fields even though the other sensors continue to update. Changing ELRS `Telem Ratio` makes no difference. 24 sensors are available when the GPS appears but If all sensors are deleted after the GPS has failed and sensors then rediscovered the GPS sensors are missing with only 19 sensors discovered. If you leave discover sensors running the GPS sensors randomly reappear but don't update just show the previous incorrect GPS data. All other telemetry fields work OK. This seems similar to other CRSF telemetry problems caused by the GPS data overloading something as soon as a fix is obtained, or maybe it's unrelated.

Any ideas ?

**Comment by OptimusTi** [comment]
No problems here. How is the gps behaving with the RX off? Sounds more like EdgeTX/Zorro issue.

**Comment by breadoven** [comment, marked as answer]
Actually my screwup. I changed the GPS "altitude" to show actual GPS altitude not take-off altitude and duplicated `crsfSerialize16(dst, altitude);` in the process. Pretty good way of breaking things. 🤦
